### PR TITLE
Immunization Controller and Service

### PR DESCRIPTION
**Pull request by sslaws**
## Status
Ready

## Fixes or Implements [AB#7898](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/7898)
- [X] Enhancement
- [ ] Bug
- [ ] Documentation

## Description:
Add logic for Immunization Controller and Service to interact with the delegate and map the data back to the front end.

## Testing
- [X] Unit Tests Updated
- [ ] Functional Tests Updated
- [ ] Not Required

### Steps to Reproduce:
If this is a bug, please provide details on how to reproduce the code.

### UI Changes
NA for this Task.

### Browsers Tested
- [ ] Chrome
- [ ] Safari
- [ ] Edge
- [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments
None

### CODE REVIEW

**Comment by healthgatewaybc** [comment]
Immunization Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=immunization_pr_332)

**Comment by healthgatewaybc** [comment]
Patient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=patientservice_pr_332)

File: Apps/Common/src/Delegates/RestPatientDelegate.cs [hunk 1]
```diff
@@ -55,46 +55,52 @@ public class RestPatientDelegate : IPatientDelegate
...
         public async Task<string> GetPatientPHNAsync(string hdid, string authorization)
         {
-            string retrievedPhn;
-            Stopwatch timer = new Stopwatch();
-            timer.Start();
-            this.logger.LogTrace($"Getting patient phn... {hdid}");
-
-            using (HttpClient client = this.httpClientService.CreateDefaultHttpClient())
+            string retrievedPhn = string.Empty;
+            Patient p = await this.GetPatientAsync(hdid, authorization).ConfigureAwait(true);
```
Comment (FuriousLlama) [inline Apps/Common/src/Delegates/RestPatientDelegate.cs]: nit: p -> patient

File: Apps/Immunization/src/Services/ImmunizationService.cs [hunk 2]
```diff
@@ -15,45 +15,92 @@
...
+            Hl7.Fhir.Model.Bundle fhirBundle = await this.immunizationDelegate.GetImmunizationBundle(request).ConfigureAwait(true);
+            IEnumerable<Hl7.Fhir.Model.Immunization> immmsLiist = fhirBundle.Entry
+                                                                            .Where(r => r.Resource is Hl7.Fhir.Model.Immunization)
+                                                                            .Select(f => (Hl7.Fhir.Model.Immunization)f.Resource);
+            foreach (Hl7.Fhir.Model.Immunization entry in immmsLiist)
             {
-                PersonalHealthNumber = "9735353315",
-                DateOfBirth = "19670602",
-            });
+                ImmunizationView iv = new ImmunizationView
```
Comment (FuriousLlama) [inline Apps/Immunization/src/Services/ImmunizationService.cs]: nit: iv => immunizationView

File: Apps/Immunization/src/Services/ImmunizationService.cs [hunk 3]
```diff
@@ -15,45 +15,92 @@
...
+                {
+                    Id = entry.Id,
+                    Name = entry.VaccineCode.Text,
+                    Status = entry.Status.ToString() !,
+                    OccurrenceDateTime = System.DateTime.Parse(entry.Occurrence.ToString() !, CultureInfo.InvariantCulture),
+                };
+                foreach (Hl7.Fhir.Model.Coding code in entry.VaccineCode.Coding)
+                {
+                    ImmunizationAgent ia = new ImmunizationAgent
```
Comment (FuriousLlama) [inline Apps/Immunization/src/Services/ImmunizationService.cs]: nit ia -> agent

**Comment by healthgatewaybc** [comment]
Immunization Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=immunization_pr_332)

**Comment by healthgatewaybc** [comment]
Patient Project [SonarQube Report](https://sonarqube-gateway.pathfinder.gov.bc.ca/dashboard?id=patientservice_pr_332)

**Pull request merged**